### ISSUE TITLE
Loss could easily be nan during training.

### ISSUE THREAD

**Issue by spacehstar** [chentuochao/Sound_Bubble#4]
Hi,

I love your idea about the sound bubble and want to do experiment locally. I followed your steps and using your synthesised data as training data and point dataset to the local downloaded synthesised data in pretrain_stage.json. I found that:

1. The grad_clip is not passed to model parameter so error will be triggered in model.  Actually, the model is suffering from gradient exploding and loss will go to nan after about 2000 iterations.
2. Even if I hard code grad_clip to 1,and after several epochs the loss still goes to nan.

May I know during training on your synthesised data, the seed number you're using? Did I do anything wrong that could let this happen?

I attached the config file I'm using. [local_pretrain_stage.json](https://github.com/user-attachments/files/19046866/local_pretrain_stage.json)

Thanks,
Fred

**Comment by chentuochao** [comment]
Hi,
From my experiences, the nan usually because the gradient vanishing happen during long sequence back propagation of LSTM. So gradient clipping cannot solve the issues. Also this NaN also happens to me randomly or hardware-specific. To alleviate it, you ca try to reduce the input sequence length to LSTM, for time domain LSTM you can reduce the input audio duration in the dataloader, for frequency-domain LSTM you can apply the frequency compression by set "conv_lstm" in json file to be True and set "lstm_down" to some number (like 2,4,5). For example [here](https://github.com/chentuochao/Sound_Bubble/blob/9581a5e2c22e9a5a9615136c07a191b7d5f73d2b/real_experiments/raspberrypi_model_finetune.json#L20C6-L20C20)

**Comment by spacehstar** [comment]
Hi,

Thanks for your such fast reply and helpful explain. I just tried to run with your configuration in your checkpoint rather than the one in https://github.com/chentuochao/Sound_Bubble/tree/main/syn_experiments, seems the training is now convergent. Notice that in the checkpoint, the loss net is changed and seems that is critical to this issue. 

Regards,
Fred
